clang/include/clang/Analysis/Analyses/LifetimeSafety/LiveOrigins.h

```diff
@@ -43,7 +43,7 @@ struct LivenessInfo {
   /// multiple uses along different paths, this will point to the use appearing
   /// earlier in the translation unit.
   /// This is 'null' when the origin is not live.
-  const UseFact *CausingUseFact;
+  const Fact *CausingFact;
```

Comment:
We can use `llvm::PointerUnion<UseFact*, OriginEscapeFact* >` to be explicit what can appear here.